Framework/API/src/MultipleFileProperty.cpp

```diff
@@ -395,6 +414,10 @@ MultipleFileProperty::setValueAsMultipleFiles(const std::string &propValue) {
     allFullFileNames.push_back(fullFileNames);
   }
 
+  // cache the new version of things
+  m_oldPropValue = propValue;
+  m_oldFoundValue = allFullFileNames;
```

Comment:
Adjust the order and you may be able to use `m_oldFoundValue = std::move(allFullFileNames)`